[docs-only] flesh out migration docs

butonic opened the pull request:
Adding more details to the migration stages with sections on:
- Why?
- User impact
- steps
- verification
- rollback
- notes
- FAQ

I'll also add TODOs where I run into missing things.

fschade commented on docs/ocis/migration.md:
```diff
@@ -20,28 +20,131 @@ The migration happens in subsequent stages while the service is online. First al
+#### Why?
+Befor letting oCIS handle end user requests we will first make it available in the internal network. By subsequently adding services we can add functionality and verify the services work as intended.
+
+Start oCIS backend and make read only tests on existing data using the `owncloudsql` storage driver which will read (and write)
```
can we mention a guide on how to get `owncloudsql` configured

fschade reviewed (commented):
good read, just a small mention

pmaier1 commented on docs/ocis/migration.md:
```diff
@@ -10,38 +10,117 @@ geekdocFilePath: migration.md
+
+### Stage 2: _introduce OpenID Connect_
+
+Basic auth requires us to properly store and manage user credentials. Something we would rather like to delegate to a tool specifically built for that task.
```
I wrote a blog post on this some time ago. Maybe you want to reference it https://owncloud.com/news/openid-connect-oidc-app/

butonic commented:
@pmaier1 I added your link. @michl19 I updated the share manage migration stage with what came to my mind.

@dragotin @micbar the TODOs feels like I wrote another roadmap ... we should link them to tickets.
